metricbeat/module/kubernetes/util/kubernetes.go

```diff
@@ -325,7 +325,7 @@ func createWatcher(
 }
 
 // addToMetricsetsUsing adds metricset identified by metricsetUsing to the list of resources using the shared watcher
-// identified by resourceName.
+// identified by resourceName. The caller of this function should be holding the lock.
```

Comment:
You have this inversed. The caller should not be holding the lock, as this function locks it.

Reply:
Yes, thank you. I fixed it now.